### PR TITLE
[Python API] Add support for blocking models

### PR DESCRIPTION
**Pull request by 4tXJ7f**
The methods for blocking models were missing in the Python API. This
commit adds the `blockModel()` and `blockModelValues()` methods as well
as the corresponding tests.

### CODE REVIEW

File: src/api/python/cvc5.pxd [hunk 1]
```diff
@@ -299,6 +299,8 @@ cdef extern from "api/cpp/cvc5.h" namespace "cvc5::api":
         bint getAbduct(const Term& conj, Term& output) except +
         bint getAbduct(const Term& conj, Grammar& grammar, Term& output) except +
         bint getAbductNext(const Term& conj) except +
+        void blockModel() except +
+        void blockModelValues(vector[Term]& terms) except +
```
Comment (nafur) [inline src/api/python/cvc5.pxd:303]: I guess this should be a `const`?

Comment (4tXJ7f) [reply src/api/python/cvc5.pxd]: Yep, done.

File: src/api/python/cvc5.pxi [hunk 2]
```diff
@@ -2315,6 +2315,43 @@ cdef class Solver:
...
 
+    def blockModel(self):
+        """
+        Block the current model. Can be called only if immediately preceded by a
+        SAT or INVALID query.
+
+        SMT-LIB:
+
+        .. code-block:: smtlib
+            (block-model)
```
Comment (nafur) [inline src/api/python/cvc5.pxi:2326]:
```suggestion
        .. code-block:: smtlib
        
            (block-model)
```

File: src/api/python/cvc5.pxi [hunk 3]
```diff
@@ -2315,6 +2315,43 @@ cdef class Solver:
...
+
+        Requires enabling option
+        :ref:`produce-models <lbl-option-produce-models>`.
+        'produce-models' and setting option
```
Comment (nafur) [inline src/api/python/cvc5.pxi:2330]:
```suggestion
        :ref:`produce-models <lbl-option-produce-models>`
        and setting option
```

File: src/api/python/cvc5.pxi [hunk 4]
```diff
@@ -2315,6 +2315,43 @@ cdef class Solver:
...
+        :ref:`block-models <lbl-option-block-models>`.
```
Comment (nafur) [inline src/api/python/cvc5.pxi:2331]:
```suggestion
        :ref:`block-models <lbl-option-block-models>`
```

File: src/api/python/cvc5.pxi [hunk 5]
```diff
@@ -2315,6 +2315,43 @@ cdef class Solver:
...
+
+        SMT-LIB:
+
+        .. code-block:: smtlib
+
+           (block-model-values ( <terms>+ ))
+
+        Requires enabling option
+        :ref:`produce-models <lbl-option-produce-models>`.
+        'produce-models'.
```
Comment (nafur) [inline src/api/python/cvc5.pxi:2349]:
```suggestion
        :ref:`produce-models <lbl-option-produce-models>`.
```

File: src/api/python/cvc5.pxi [hunk 6]
```diff
@@ -2315,6 +2315,44 @@ cdef class Solver:
...
```
Comment (nafur) [inline src/api/python/cvc5.pxi:2350]:
```suggestion
```

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**